docs/Optimizations.md

```diff
@@ -67,6 +67,12 @@ Below you can see the list of currently supported graph optimizations:
     This optimization performs a classic CSE with a goal of avoiding of any
     results that were computed already.
 
+  * Optimization of ReduceMean nodes
```

Comment:
Why is this an optimization? Why would one form be better?

Reply:
At least in our case, Reduce Mean is implemented using Average Pool. Originally I considered having it completely controlled by the backend, but based on @jfix71 's   comments in #2343, we took the recommended, shorter route and considered this as an always-to-do optimization.

Reply:
I consider it an optimization because otherwise we lower `BatchedReduceMean` to multiple nodes -- i.e. it's more efficient to execute an `AvgPool` than to lower it/execute it via a `BatchedReduceAdd` followed by a `Div`. 

But it really is dependent on what the backend supports -- if a backend has an efficient `BatchedReduceMean` implementation then they would likely prevent lowering on the node and add a backend-specific `BatchedReduceMeanInst`. At that point perhaps they wouldn't want this `ReduceMean -> AvgPool` optimization enabled. But this just another area where we'd want better backend control over graph optimizations (see https://github.com/pytorch/glow/issues/1641).